WordPress/gutenberg
# [RNMobile] RangeControl markup

## Description

PR is connected with [#1449](https://github.com/wordpress-mobile/gutenberg-mobile/issues/1449) in gutenberg-mobile.

Please also refer to:

[Related gutenberg-mobile PR](https://github.com/wordpress-mobile/gutenberg-mobile/pull/1471)

It presents ability to select all parents in cascade way (according to the way how selecting in groups works on web version)


## How has this been tested?




- Add below range cell implementation inside `InspectorControls` in `edit.native.js` 
<details>
<summary>Usage of RangeControl </summary>

```
<RangeControl
icon={ 'admin-settings' }
label={ __( 'Slider' ) }
value={ sliderVal }
initialPosition={ 25 }
min={ 20 }
max={ 100 }
separatorType={ 'fullWidth' }
onChange={ this.updateSlider }
/>
```
</details>

- add handler to change slider value
<details>
<summary>Handler to change slider value</summary>

```
updateSlider( newVal ) {
    this.props.setAttributes( { sliderVal: newVal } );
}
```
</details>

- reset `sliderVal` to `null` in `onClearSettings()` function

I have also test this feature by merging this PR changes and [this branch](https://github.com/WordPress/gutenberg/pull/18157) into [Gallery try feature](https://github.com/WordPress/gutenberg/pull/18111). 

After switch to Gallery branch I need to temporary comment some code  in `edit.js` in `block-library/gallery` which produced error. See details below:

<details>

	componentDidMount() {
		const { attributes, mediaUpload } = this.props;
		const { images } = attributes;
		if ( every( images, ( { url } ) => isBlobURL( url ) ) ) {
			const filesList = map( images, ( { url } ) => getBlobByURL( url ) );
			forEach( images, ( { url } ) => revokeBlobURL( url ) );
			// mediaUpload( {
			// 	filesList,
			// 	onFileChange: this.onSelectImages,
			// 	allowedTypes: [ 'image' ],
			// } );
		}
	}

</details>

Then I was able to test if passing mobile specific props do not breaks anything in web and if web props do not break anything in mobile.

## Screenshots 

<image src="https://user-images.githubusercontent.com/21242757/67663042-e2e1b780-f964-11e9-82be-b209ec8d3615.png" width="300" />

## Types of changes




Adds new layer on top of `BottomSheet.RangeCell` to use it in the same way as Sidebar `RangeContol` slider in web version.

## Checklist:
